neurolang/relational_algebra_provenance.py

```diff
@@ -218,14 +228,76 @@ def prov_concatenate_constant_column(self, concat_op):
         )
         return ProvenanceAlgebraSet(self.walk(relation).value, new_prov_col)
 
+    @add_match(IndependentProjection)
+    def prov_independent_projection(self, proj_op):
+        """
+        An independent projection assumes tuples to be probabilistically
+        independent.
+
+        Thus, the marginal probability of two tuples being true in the same
+        possible world is equal to the complement of their probability of being
+        false in the same possible world. In other words, for a relation R and
+        two tuples (t1, t2), the marginal probability
+
+            P[R(t1) & R(t2)]
+
+        of them being true at the same time is equal to
+
+            1 - (1 - P[R(t1)]) (1 - P[R(t2)])
+
+        This formula corresponds to the sometimes called 'noisy-or' formula.
+
+        """
+        prov_set = self.walk(proj_op.relation)
+        agg_fun = lambda p: 1 - numpy.prod(1 - p)
```

Comment:
This way of implementing the aggregation will be very slow. Can you implement it such that it's translated natively to SQL operations? As with a string or something else?